notifier/src/main/kotlin/modules/JiraNotifier.kt

```diff
@@ -0,0 +1,140 @@
+import com.atlassian.jira.rest.client.api.domain.Comment
+import com.atlassian.jira.rest.client.api.domain.IssueType
+import com.atlassian.jira.rest.client.api.domain.input.IssueInputBuilder
+import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory
+
+import java.net.URI
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+
+import org.ossreviewtoolkit.model.Failure
+import org.ossreviewtoolkit.model.Result
+import org.ossreviewtoolkit.model.Success
+import org.ossreviewtoolkit.model.config.JiraConfiguration
+import org.ossreviewtoolkit.utils.collectMessagesAsString
+import org.ossreviewtoolkit.utils.log
+
+class JiraNotifier(
+    val config: JiraConfiguration,
+    private val restClient: JiraRestClient = AsynchronousJiraRestClientFactory()
+        .createWithBasicHttpAuthentication(URI(config.host), config.username, config.password)
+) {
+    /** Concurrent map of all available issue types for each project */
+    private val projectIssueTypes: ConcurrentMap<String, Map<String, IssueType>> = ConcurrentHashMap()
+
+    /**
+     * A Wrapper function in order to create an issue in the given Jira instance using the Jira REST API.
+     *
+     * @param projectKey The project key of the project in Jira in which the issue should be created.
+     * @param summary The summary of the issue.
+     * @param description The description of the issue.
+     * @param issueType The issue type of the issue. The given type has to be specified in the given project in Jira.
+     * @param assignee The assignee name of the issue. The given name has to be present in the Jira instance.
+     * @param avoidDuplicates If true (default), the function will not create a new issue if there is already exactly
+     *                        one issue with the same summary, but it will add a comment to the issue with the given
+     *                        description. If false, it will always create a new issue even if there are issues with the
+     *                        same summary.
+     *
+     * @return Return a [Result] that encapsulates the created issue on success, or the caught exception on failure.
+     */
+    fun createIssue(
+        projectKey: String,
+        summary: String,
+        description: String,
+        issueType: String,
+        assignee: String? = null,
+        avoidDuplicates: Boolean = true
+    ): Result<BasicIssue> {
+        if (!projectIssueTypes.containsKey(projectKey)) {
+            projectIssueTypes[projectKey] = restClient.projectClient.getProject(projectKey).claim()
+                .issueTypes
+                .map { it.name to it }
+                .toMap()
+        }
+
+        if (!isIssueTypeValid(projectKey, issueType)) {
+            return Failure("The issue type '$issueType' is not valid. Use a valid issue type, specified in " +
+                    "your project '$projectKey'")
+        }
+
+        val issueInputBuilder = IssueInputBuilder()
```

Comment:
I am not familiar with the JIRA API, so just a question: As far as I know, JIRA can be configured to support many more properties for issues. Is this somehow reflected in the API, e.g. via generic properties? If so, should this be supported by the notifier as well?

Reply:
IMO we should keep the notifier as simple as possible, if there is a need for this, we could add it later.